File: changedetection/FileToString.java
// TODO Add Unittest

import java.io.*;

public class FileToString {

  /**
   * Returns a String containing all the characters from the file at filePath, with all the newlines
   * and spaces removed.
   *
   * @param filePath : String
   * @return String/null
   */
  public static String convertToString(String filePath) {
    File file = new File(filePath);
    StringBuilder s = new StringBuilder();
    try {
      BufferedReader in = new BufferedReader(new FileReader(file));
      String buf = in.readLine();
      while (buf != null) {
        if (!(buf.trim().equals(""))) {
          s.append(buf.trim());
          s.append(" ");
        }
        buf = in.readLine();
      }
      return s.toString();
    } catch (IOException iox) {
      iox.printStackTrace();
    }
    return null;
  }

  public static void main(String[] args) {
    String filePath = "/* Bring up Dialog Potentially to get file or use cwd */";
    String res = convertToString(filePath);
  }
}

File: changedetection/FileToString.kt
// TODO add unittest, research Kotlin Test Frameworks

import java.io.*

object FileToString {
    /**
     * Returns a String containing all the characters from the file at filePath, with all the newlines and spaces
     * removed.
     *
     * @param filePath : String
     * @return String/null
     */
    fun convertToString(filePath: String): String? {
        val file = File(filePath)
        val s = StringBuilder()
        try {
            val `in` = BufferedReader(FileReader(file))
            var buf: String? = `in`.readLine()
            while (buf != null) {
                if (buf.trim { it <= ' ' } != "") {
                    s.append(buf.trim { it <= ' ' })
                    s.append(" ")
                }
                buf = `in`.readLine()
            }
            return s.toString()
        } catch (iox: IOException) {
            iox.printStackTrace()
        }

        return null
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val filePath = "/* Bring up Dialog Potentially to get file or use cwd */"
        val res = convertToString(filePath)
    }
}
